components/Players/PlayerManager.vue

```diff
@@ -193,12 +206,42 @@ export default Vue.extend({
         default:
           return this.getCurrentItem.Name;
       }
+    },
+    timeLeft(): number {
+      return parseInt((this.mediaDuration - this.getCurrentTime).toFixed());
+    },
+    mediaDuration(): number {
+      // In seconds
+      return this.ticksToMs(this.getCurrentItem.RunTimeTicks) / 1000;
+    },
+    showUpNext(): boolean {
+      if (
+        this.isMinimized ||
+        this.upNextUserHidden ||
+        this.getCurrentItem.Type !== 'Episode' ||
+        !this.getNextItem
+      ) {
+        return false;
+      }
+      // How much of the video to go through before showing
+      // the 'up-next' component
+      const showAtProgressPercentage = 0.97;
+
+      if (
+        this.getCurrentTime <=
+        this.mediaDuration * showAtProgressPercentage
+      ) {
+        return false;
+      }
+      return true;
     }
   },
   created() {
     this.$store.subscribe((mutation, state: AppState) => {
       switch (mutation.type) {
         case 'playbackManager/INCREASE_QUEUE_INDEX':
+          this.upNextUserHidden = false;
+          break;
```

Comment:
What if I go back to the previous item while the popup is shown? Will the warning be persistent, or it will be removed because the percentage completed is less than 97%?

Reply:
If you go back it would've stayed hidden even at the last 97%. Fixed this.